pepper-apis/src/main/java/org/broadinstitute/ddp/export/collectors/ActivityInstanceSelectQuestionFormatStrategy.java

```diff
@@ -38,13 +63,36 @@
 
     @Override
     public List<String> headers(ActivityInstanceSelectQuestionDef definition) {
-        return Arrays.asList(definition.getStableId());
+        return Collections.singletonList(definition.getStableId());
     }
 
     @Override
     public Map<String, String> collect(ActivityInstanceSelectQuestionDef question, ActivityInstanceSelectAnswer answer) {
+        return collect(question, answer, null);
+    }
+
+    public Map<String, String> collect(ActivityInstanceSelectQuestionDef question, ActivityInstanceSelectAnswer answer, Long userId) {
         Map<String, String> record = new HashMap<>();
-        record.put(question.getStableId(), StringUtils.defaultString(answer.getValue(), ""));
+        String name;
+        if (userId != null && studyGuid != null) {
+            name = TransactionWrapper.withTxn(TransactionWrapper.DB.APIS, handle -> {
+                User user = handle.attach(UserDao.class).findUserById(userId).orElseThrow();
+                List<ActivityInstanceSummary> summaries = activityInstanceService.findTranslatedInstanceSummaries(
+                        handle, user.getGuid(), studyGuid, Set.copyOf(question.getActivityCodes()),
+                        LanguageStore.getDefault().getIsoCode()).stream()
+                        .filter(i -> i.getActivityInstanceGuid().equals(answer.getValue())).collect(Collectors.toList());
+                if (CollectionUtils.isNotEmpty(summaries)) {
+                    Map<String, FormResponse> responses = activityInstanceService.countQuestionsAndAnswers(
+                            handle, user.getGuid(), user.getGuid(), studyGuid, summaries);
+                    activityInstanceService.renderInstanceSummaries(handle, user.getId(), user.getGuid(), studyGuid, summaries, responses);
+                    return summaries.get(0).getActivityName();
+                }
+                return answer.getValue();
+            });
+        } else {
```

Comment:
studyGuid will not be null. Trying to understand under if userId will be null and If it does , what are the scenarios where userId can be null ? 
In these cases (else), are we using the answer guid?

Reply:
We use an answer value in these cases (in other words, selected instance GUID). The reason why I made these checks is that  the nulls still could be passed at some point of time even though we don't expect that. So my intention was to still maintain a minimal functionality instead of failing the whole export.